[FIX] website: include archived snippet views to avoid duplication error

apan-odoo opened the pull request:
- While calling `_generate_primary_snippet_templates`, `create_missing_views` fetches the existing snippet views. However, archived snippet is ignored, causing `create_missing_views` to attempt to create the snippet again, mistakenly considering it as a missing view. This leads to a traceback:

```py
  File "/data/build/odoo/odoo/sql_db.py", line 332, in execute
    res = self._obj.execute(query, params)
psycopg2.errors.UniqueViolation: duplicate key value violates unique constraint "ir_model_data_module_name_uniq_index"
DETAIL:  Key (module, name)=(website, new_page_template_services_0_s_website_form) already exists.
```

- To prevent such errors, it's better to fetch inactive views as well during creation.

opw-4510585
upg-2452055



---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/197107.png)](https://mergebot.odoo.com/odoo/odoo/pull/197107)

apan-odoo commented:
Hello @odoo/rd-website team,
Can you please have a look at this PR?
Thanks!

apan-odoo commented:
Hello @odoo/rd-website team, Can you please review this PR. Thank you 🐱

qsm-odoo reviewed (changes requested):
Hello, thanks for the fix.
- Is it a theoretical case where the user went and disabled random views in its database? Or is there a real flow leading to this case? If there is a flow, add it in the commit message :pray: 
- Commit title: "when applied, this commit will _fetch inactive views_" -> I have no clue what the fix is about, a better title should be found
- Max 72 characters per line in commit messages (of course not required for the error log but for the rest, yes)
- `opw-XXX`, not `OPW - XXX` -> not adding spaces and adding a `-` is the whole point: we don't want XXX being mistakenly considered as a commit hash or something else by GitHub in the future (it would transform it into a link which is unrelated). And in general, we don't uppercase it.
- Is it the right OPW number? It seems closed and not mentioning the same error log.
- What's UPG ? :thinking:

apan-odoo commented:
> Hello, thanks for the fix.
> 
> * Is it a theoretical case where the user went and disabled random views in its database? Or is there a real flow leading to this case? If there is a flow, add it in the commit message 🙏
> * Commit title: "when applied, this commit will _fetch inactive views_" -> I have no clue what the fix is about, a better title should be found
> * Max 72 characters per line in commit messages (of course not required for the error log but for the rest, yes)
> * `opw-XXX`, not `OPW - XXX` -> not adding spaces and adding a `-` is the whole point: we don't want XXX being mistakenly considered as a commit hash or something else by GitHub in the future (it would transform it into a link which is unrelated). And in general, we don't uppercase it.
> * Is it the right OPW number? It seems closed and not mentioning the same error log.
> * What's UPG ? 🤔

Hello @qsm-odoo thank you for the in-depth review!
- I didn't find any real flow on how the user archived the snippet views. Disabling it manually is the only method I'm aware of.
- I have updated the commit title, I hope now it make more sense what this commit addresses.
- Updated the commit message lines to not exceed more than 72 chars.
- Noted and corrected the OPW part. Also, the OPW is correct—please scroll down to find the pad link. The OPW is closed because I provided a specific solution to the client using a migration script.
- UPG stands for upgrade request which can be seen at https://upgrade.odoo.com/
